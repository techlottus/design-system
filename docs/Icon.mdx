---
name: Icon
menu: Components
---

import { Props } from 'docz'
import Icon from '../packages/Icon';
import Nav from '../packages/Nav';


# Icon
Icon component docs

## Props
<Props of={Icon}/>

## Basic
```jsx
<Icon size={32} icon="icon-twitter" />
<Icon size={32} icon="icon-facebook" />
<Icon size={32} icon="icon-linkedin" />
<Icon size={32} icon="icon-instagram" />
```
<Nav>
  <Nav.Item>
    <Icon size={32} icon="icon-twitter" />
  </Nav.Item>
  <Nav.Item>
    <Icon size={32} icon="icon-facebook" />
  </Nav.Item>
  <Nav.Item>
    <Icon size={32} icon="icon-linkedin" />
  </Nav.Item>
  <Nav.Item>
    <Icon size={32} icon="icon-instagram-glyph" />
  </Nav.Item>
</Nav>

## With gap
```jsx
<Icon size={32} gap={2} icon="icon-arrows" />
<Icon size={32} gap={4} icon="icon-flag" />
<Icon size={32} gap={6} icon="icon-money" />
<Icon size={32} gap={8} icon="icon-cart" />
```
<Nav>
  <Nav.Item>
    <Icon size={32} gap={2} icon="icon-arrows" />
  </Nav.Item>
  <Nav.Item>
    <Icon size={32} gap={4} icon="icon-flag" />
  </Nav.Item>
  <Nav.Item>
    <Icon size={32} gap={6} icon="icon-money" />
  </Nav.Item>
  <Nav.Item>
    <Icon size={32} gap={8} icon="icon-cart" />
  </Nav.Item>
</Nav>