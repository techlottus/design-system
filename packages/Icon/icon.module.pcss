.icon {
  &Wrapper {
    @apply flex justify-center items-center;
  }
}