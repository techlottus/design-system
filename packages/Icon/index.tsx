import * as React from 'react';
import { IProps } from '../common/props';
import classNames from 'classnames';
import style from './icon.module.pcss';

export interface IIconProps extends IProps {
  size?: number;
  gap?: number;
  icon?: string;
}

const Icon: React.SFC<IIconProps> = ({ size, gap, className, icon }) => {
  const classes = classNames(className, 'betomic', icon);
  return (<div className={style.iconWrapper} style={{
    width: size,
    height: size,
    padding: gap,
  }}>
    <i style={{
      fontSize: (size - gap),
    }} className={classes}></i>
  </div>);
};

Icon.defaultProps = {
  size: 16,
  gap: 0,
};

export default Icon;
