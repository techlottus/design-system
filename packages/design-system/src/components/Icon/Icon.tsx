import cn from "classnames";
import { IconType } from "../Types/Icon.types";

const Icon: React.FC<IconType> = (props: IconType) => {
  const { iconName, className = "", variant = "solid" } = props;
  return (
    <span
      className={cn(className, {
        ["font-icons-outlined"]: variant === "outlined",
        ["font-icons-solid"]: variant === "solid",
      })}
    >
      <i>{iconName}</i>
    </span>
  );
};

export default Icon;
