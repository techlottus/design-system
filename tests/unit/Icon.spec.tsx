import * as React from 'react';
import { shallow } from 'enzyme';
import Icon from '../../packages/Icon/index';

describe('Icon.tsx', () => {
  it('should match the snapshot', () => {
    const vm = shallow(<Icon size='14px' className={'fab fa-amazon text-neutral'} />);
    expect(vm).toMatchSnapshot();
    expect(vm.find('i').hasClass('fab fa-amazon text-neutral')).toBe(true);
    expect(vm.find('i').hasClass('fab fa-amazon text-warning')).toBe(false);
    expect(vm.props().children.props.className).toEqual('fab fa-amazon text-neutral');
    expect(vm.props().children.props.style).toEqual({fontSize: '14px'});
    console.log('props: ', vm.props());
  });
});
