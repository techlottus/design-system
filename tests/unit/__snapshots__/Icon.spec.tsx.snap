// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon.tsx should match the snapshot 1`] = `
<div>
  <i
    className=" text-neutral fab fa-amazon"
    style={
      Object {
        "fontSize": "12px",
      }
    }
  />
</div>
`;
